frontend/src/_includes/partials/colored-badges.html

```diff
@@ -0,0 +1,53 @@
+<div class="mb-1 flex items-center space-x-1">
+  <!-- Percentage -->
+  <div
+    class="relative overflow-hidden rounded-lg border border-gray-300"
+    title="
+      {%- if audioObj.percentage > 0 -%}
+        How much Srila Gurudeva is speaking in the recording
+      {%- else -%}
+        Only other guru-varga is speaking in the recording
+      {%- endif -%}
+    "
+  >
+    <div
+      class="absolute z-auto h-full w-[var(--percentage)] bg-gray-300 shadow-none"
+      style="--percentage: {% if audioObj.percentage > 0 -%}{{ audioObj.percentage | percentage }}{%- else -%}100%{%- endif -%}"
+    ></div>
+    <div class="w-full whitespace-nowrap px-1.5 pb-[1px] opacity-95">
+      {%- if audioObj.percentage > 0 -%}
+        Srila Gurudeva {{ audioObj.percentage | percentage }}
+      {%- else -%}
+        Other guru-varga
+      {%- endif -%}
+    </div>
+  </div>
```

Comment:
Let's split this into two partials, each will get only required value, not the whole audio object.